### ISSUE TITLE
Custom labels in dynamic modeling and topic per class

### ISSUE THREAD

**Issue by sercankiyak** [MaartenGr/BERTopic#2154]
### Have you searched existing issues?  🔎

- [X] I have searched and found no existing issues

### Desribe the bug

Hi,

I’ve come across a small issue that’s left me a bit puzzled. I’m not entirely sure if it’s a bug, but it does seem confusing. When I run topics_per_class with my topic model:

topics_per_class = topic_model.topics_per_class(docs, classes=classes, global_tuning=False)

The resulting topics_per_class looks like this:

Topic | Words | Frequency | Class
-1 | jaar, mensen, zegt, vrouwen, politie | 1421 | nl

This has no custom_labels. That is not ideal but OK. 

But when I run this 

fig = topic_model.visualize_topics_per_class(topics_per_class, custom_labels=True)

Then rerun topics_per_class in my jupyter, I get now this: 

  | Topic | Words | Frequency | Class | Name
-1 | jaar, mensen, zegt, vrouwen, politie | 1421 | nl | -1_mensen_vrouwen_politie_kinderen

Now the "Name" is added. It’s unclear why it isn’t already included or why there’s no parameter like custom_labels=True when creating topics_per_class. I checked the [documentation](https://maartengr.github.io/BERTopic/api/bertopic.html#bertopic._bertopic.BERTopic.topics_per_class), and this parameter doesn’t exist, and attempting to use it gives an error in the first step.

I just wanted to bring this up. As title indicates, I’ve also experienced the same issue with dynamic topic modeling as well. 

I sometimes find the table version easier to use, so I’m curious why it works like this, or maybe I’m missing something?

This is only my second time reporting something (on GitHub in general), so I hope I explained it clearly. Thanks again for everyone’s work and effort on BERTopic.

Best,

### Reproduction

```python
from bertopic import BERTopic

topic_model = BERTopic(verbose=True)
topics, probs = topic_model.fit_transform(docs)

topics_per_class = topic_model.topics_per_class(docs, classes=classes)

print(topics_per_class.columns)
# No custom labels at this step

topic_model.visualize_topics_per_class(topics_per_class, top_n_topics=10)

print(topics_per_class.columns )
# Now custom labels are the last column in the df.
```

### BERTopic Version

0.16.2

**Comment by MaartenGr** [comment]
This might be because `.visualize_topics_per_class` does not create a copy of `topics_per_class` but instead directly affects the variable. I believe that's why `topics_per_class` gets that additional column after running that function. It should an easy fix by simply creating a copy of the dataframe before attempting to update it.

**Comment by sercankiyak** [comment]
Thanks for the explanation. I understand your suggestion from a programming point of view. However, as a user, I would prefer there was parameter in topics_per_class(), like custom_labels=TRUE. Then, one could create the table already with the custom labels. I would prefer that because I think analyzing the tables is much better than the visual. I can highlight things, take notes etc. Would adding such a parameter be a bad idea, if so why? Or maybe there is a way that I did not figure out yet. Or what is the proper method to create a table with the custom labels while using dynamic modeling or topic per class?

**Comment by sercankiyak** [comment]
Thanks!

**Comment by ShivamSrng-OIE** [comment]
Hi, I checked this against BERTopic 0.17.4 and the current master branch. Both `visualize_topics_per_class` and `visualize_topics_over_time` still add the `Name` column directly to the supplied DataFrame.

I would like to address this by creating a defensive copy inside both visualization functions and adding regression tests confirming that the original DataFrames are not mutated for default, custom, and aspect-based labels.

Since the issue has been open for some time, could you confirm that this is still the preferred scope? If so, I would be happy to submit a PR.
